app/src/main/java/com/example/mystore/ui/login/LoginScreen.kt

```diff
@@ -129,18 +136,38 @@ fun LoginScreen(navigateToProductDetailsScreen: () -> Unit = {}) {
             modifier = Modifier.fillMaxWidth(),
             onClick = {
                 isValidInput = validateFields(emailField, passwordField)
-                if (isValidInput && emailField == rightEmail && passwordField == rightPassword) {
-                    navigateToProductDetailsScreen()
+                if (isValidInput) {
+                    isButtonEnabled = false
+                    val loginModel = LoginModel(emailField, passwordField)
+                    coroutineScope.launch(Dispatchers.IO) {
```

Comment:
`coroutineScope`s are mostly used when we don't have a view model. The purpose of the view model is to be responsible for the whole business logic, including the whole request process. It can facilitate its handy `viewModelScope` to handle `suspend` functions and modify various local (to the view model itself) states from inside the function (`login()` in this particular case).

Your composable view here ideally should do nothing but call `loginViewModel.login(loginModel)`. Then it can subscribe to various result/error states that you have in the view model via a `LaunchedEffect` and either navigate or display a snackbar/toast inside the `LaunchedEffect` itself.

We haven't shown `LaunchedEffect` yet, though, so another viable option is to pass in a callback function as a parameter to `loginViewModel.login(loginModel)` and handle the results inside.